osquery/tables/system/windows/shimcache.cpp

```diff
@@ -0,0 +1,216 @@
+
+auto parseShimcacheData(std::string data,
+                        size_t pos,
+                        std::string token,
+                        std::string execution_flag_exists,
+                        std::string delimter) {
+  ShimcacheData shimcache;
+  std::string path_length = token.substr(16, 4);
+
+  // swap endianess
+  std::reverse(path_length.begin(), path_length.end());
+
+  for (std::size_t i = 0; i < path_length.length(); i += 2) {
+    char temp = path_length[i];
+    path_length[i] = path_length[i + 1];
+    path_length[i + 1] = temp;
+  }
+  int shimcache_file_path = std::stoi(path_length, nullptr, 16);
+
+  // Registry data is in Unicode (extra 0x00)
+  std::string path = token.substr(20, shimcache_file_path * 2);
+  boost::erase_all(path, "00");
+  // Windows Store entries have extra data, the extra data includes tabs in the
+  // entry (Unicode value 09). Replace any with spaces due to table formatting
+  // issues
+  boost::replace_all(path, "09", "20");
+  std::string string_path;
+
+  // Convert hex path to readable string
+  try {
+    string_path = boost::algorithm::unhex(path);
+  } catch (boost::algorithm::hex_decode_error) {
+    LOG(WARNING) << "ERROR: Failed to decode Shimcache hex values to string";
+    shimcache.last_modified = 0LL;
+    return shimcache;
+  }
+
+  shimcache.path = string_path;
+  int shimcache_modified_start = 0;
+  int execution_flag_start = 0;
+
+  // If execution flag exists set where the flag starts in the substring
+  if (execution_flag_exists == "true") {
+    shimcache_modified_start = 40;
+    execution_flag_start = 24;
+  } else {
+    shimcache_modified_start = 20;
+  }
+  std::string shimcache_time =
+      token.substr(shimcache_modified_start + shimcache_file_path * 2, 16);
+
+  // Check to make sure the shimcache entry is greater than zero
+  if (string_path.length() > 0) {
+    // Get last modified time of the file
+    shimcache.last_modified = lastExecute(shimcache_time);
+  } else {
+    shimcache.last_modified = 0LL;
+  }
+
+  if (execution_flag_exists == "true") {
+    int shimcache_flag = std::stoi(
+        token.substr(execution_flag_start + shimcache_file_path * 2, 2),
+        nullptr,
+        16);
+    // Perform Bitwise AND to determine TRUE or FALSE
+    if (shimcache_flag & 2) {
+      shimcache.execution_flag = "true";
+    } else {
+      shimcache.execution_flag = "false";
+    }
+  }
+  return shimcache;
+}
+
+auto parseRegistry(std::set<std::string> shimcacheControl,
+                   Row r,
+                   QueryData results) {
+  for (const auto& rKey : shimcacheControl) {
+    std::size_t shimcache_entry = rKey.find(
+        "Control\\Session "
+        "Manager\\AppCompatCache");
+
+    if (shimcache_entry != std::string::npos) {
+      QueryData shimcacheResults;
+      int entry = 1;
+
+      queryKey(rKey, shimcacheResults);
+      for (const auto& aKey : shimcacheResults) {
+        if (aKey.at("name") == "AppCompatCache") {
```

Comment:
Use the "return early" pattern mentioned above:

```cpp
if (aKey.at("name") != "AppCompatCache") {
  continue;
}
```

Reply:
Switched to return early